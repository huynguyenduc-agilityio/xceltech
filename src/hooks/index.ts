export * from './useToast';
export * from './useConfirm';
export * from './useAuth';
export * from './useUser';
export * from './useLeave';
export * from './useAccount';
export * from './usePagination';
export * from './useNotification';
export * from './useJob';
export * from './useDocument';
export * from './useEducation';
export * from './useGuarantor';
export * from './useFamily';
export * from './useFinancial';
