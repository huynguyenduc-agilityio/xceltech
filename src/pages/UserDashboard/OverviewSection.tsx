// Icons
import { CalendarIcon } from '@/icons';

// Constants
import { PROGRESS_LEAVE_ITEMS } from '@/__mocks__';

// Mocks
import { mockPaySlipData } from '@/__mocks__';

// Components
import { Button, OverviewCard, Progress } from '@/components';

const OverviewSection = () => {
  return (
    <div className="grid grid-cols-1 md:grid-cols-2 gap-10 mt-[64px]">
      {/* Available Leave Days */}
      <OverviewCard title="Available Leave Days">
        <div className="flex flex-col gap-7">
          {PROGRESS_LEAVE_ITEMS.map(({ key, title, value }) => (
            <Progress
              key={key}
              aria-label={`Progress ${value}`}
              value={value}
              indicatorContent={title}
            />
          ))}
        </div>
      </OverviewCard>

      {/* Birthdays */}
      <OverviewCard title="Birthdays">
        <div className="flex flex-col gap-[14px]">
          {[...Array(5)].map((_, idx) => (
            <div
              key={idx}
              className="flex items-center justify-between px-[30px] py-3 bg-blue-light rounded-regular"
            >
              <div className="flex items-center gap-6">
                <CalendarIcon
                  width={22}
                  height={25}
                  style={{ fill: 'black' }}
                  className="text-black-default"
                />
                <span className="text-lg text-gray-steel">
                  Biruk Kidan's Day - April 25th
                </span>
              </div>
              <Button
                size="sm"
                variant="secondary"
                className="w-[182px] rounded-regular text-base"
              >
                Send Wishes
              </Button>
            </div>
          ))}
        </div>
      </OverviewCard>

      {/* Pay Slip Breakdown */}
      <OverviewCard title="April Pay Slip Breakdown">
        <div className="grid grid-cols-4 text-lg bg-blue-light text-gray-steel px-7 py-4 rounded-regular">
          <div>Earnings</div>
          <div>Amount</div>
          <div>Deductions</div>
          <div>Total</div>
        </div>

        {mockPaySlipData.map((row) => (
          <div
            key={row.type}
            className="grid grid-cols-4 text-lg bg-blue-light text-gray-steel px-7 py-4 rounded-regular mt-3"
          >
            <div>{row.type}</div>
            <div>{row.amount}</div>
            <div>{row.deductions}</div>
            <div>{row.total}</div>
          </div>
        ))}
      </OverviewCard>
    </div>
  );
};

export default OverviewSection;
