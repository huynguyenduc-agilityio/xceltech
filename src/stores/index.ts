export * from './dialog';
export * from './authentication';
