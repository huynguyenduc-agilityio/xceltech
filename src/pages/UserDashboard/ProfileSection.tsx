import { useNavigate } from 'react-router-dom';

// Constants
import { IMAGES, USER_PAGE } from '@/constants';

// Hooks
import { useGetInfoUser } from '@/hooks';

// Utils
import { getInitialsAvatar } from '@/utils';

// Components
import { Avatar, Button } from '@/components';

const ProfileSection = () => {
  const { userInfo } = useGetInfoUser();
  const navigate = useNavigate();

  const { firstName = '', lastName = '', avatar = '', job } = userInfo || {};

  return (
    <div className="relative bg-blue-midnightAzure text-white mt-8 pl-[52px] py-[52px] flex items-center justify-between rounded-regular">
      <div className="flex items-center gap-9">
        <div className="w-[139px] h-[139px] p-2 rounded-full border-2 border-white">
          <Avatar
            src={avatar as string}
            fallback={getInitialsAvatar(`${firstName} ${lastName}`)}
            size="100%"
            className="text-2xl"
          />
        </div>
        <div className="flex flex-col gap-4">
          <span className="text-3xl font-bold">
            {`${firstName} ${lastName}`}
          </span>
          <span className="text-2xl">{job?.name}</span>
        </div>
      </div>
      <div className="flex items-center gap-8">
        <Button
          variant="secondary"
          className="w-[252px] h-[65px] text-xl"
          onClick={() => navigate(USER_PAGE.PROFILE_EDIT)}
        >
          Edit Profile
        </Button>

        <img alt="profile-icon" src={IMAGES.ARROW} />
      </div>
    </div>
  );
};

export default ProfileSection;
