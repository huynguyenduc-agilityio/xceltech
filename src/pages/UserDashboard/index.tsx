import { Link } from 'react-router-dom';

// Constants
import { USER_PAGE } from '@/constants';

// Hooks
import { useGetInfoUser } from '@/hooks';

// Utils
import { getInitialsAvatar } from '@/utils';

// Components
import { Avatar, Button } from '@/components';

import OverviewSection from './OverviewSection';
import QuickAction from './QuickAction';

const UserDashboard = () => {
  const { userInfo } = useGetInfoUser();

  const { firstName = '', lastName = '', avatar = '', job } = userInfo || {};

  return (
    <>
      <h1 className="text-2xl text-primary font-bold">Dashboard</h1>
      {/* Profile Section */}
      <div className="relative bg-blue-midnightAzure text-white mt-8 pl-[52px] py-[52px] flex items-center justify-between rounded-regular">
        <div className="flex items-center gap-9">
          <div className="w-[139px] h-[139px] p-2 rounded-full border-2 border-white">
            <Avatar
              src={avatar as string}
              fallback={getInitialsAvatar(`${firstName} ${lastName}`)}
              size="100%"
              className="text-2xl"
            />
          </div>
          <div className="flex flex-col gap-4">
            <span className="text-3xl font-bold">
              {`${firstName} ${lastName}`}
            </span>
            <span className="text-2xl">{job?.name}</span>
          </div>
        </div>
        <div className="flex items-center gap-8">
          <Link to={USER_PAGE.PROFILE_EDIT}>
            <Button variant="secondary" className="w-[252px] h-[65px] text-xl">
              Edit Profile
            </Button>
          </Link>

          <img alt="profile-icon" src="/assets/images/arrow.webp" />
        </div>
      </div>

      {/* Quick Actions */}
      <QuickAction />

      {/* Grid Layout */}
      <OverviewSection />
    </>
  );
};

export default UserDashboard;
