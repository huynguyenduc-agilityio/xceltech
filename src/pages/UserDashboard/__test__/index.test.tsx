import { ReactNode } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

// Hooks
import { useGetInfoUser } from '@/hooks';

// Stores
import { useUser } from '@/stores';

import UserDashboard from '..';

jest.mock('@/hooks', () => ({
  useGetInfoUser: jest.fn(),
}));

jest.mock('@/stores', () => ({
  useUser: jest.fn(),
}));

const renderWithRouter = (children: ReactNode) => {
  return render(<MemoryRouter>{children}</MemoryRouter>);
};

describe('UserDashboard', () => {
  beforeEach(() => {
    (useUser as jest.Mock).mockReturnValue({ userId: '123' });
    (useGetInfoUser as jest.Mock).mockReturnValue({
      userInfo: {
        firstName: 'John',
        lastName: 'Doe',
        avatar: '/avatar.jpg',
        job: {
          name: 'Software Engineer',
        },
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly and matches snapshot', () => {
    const { asFragment } = renderWithRouter(<UserDashboard />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('handles undefined userInfo gracefully', () => {
    (useGetInfoUser as jest.Mock).mockReturnValue({
      userInfo: undefined,
    });

    renderWithRouter(<UserDashboard />);

    const spans = screen.queryAllByText((content, element) => {
      return element?.tagName.toLowerCase() === 'span' && content.trim() === '';
    });

    expect(spans.length).toBeGreaterThan(0);
  });

  it('handles empty userInfo object', () => {
    (useGetInfoUser as jest.Mock).mockReturnValue({
      userInfo: {},
    });

    const { container } = renderWithRouter(<UserDashboard />);

    expect(container).toBeInTheDocument();
    expect(screen.queryByText('Software Engineer')).not.toBeInTheDocument();
  });

  it('displays user information correctly', () => {
    renderWithRouter(<UserDashboard />);
    expect(screen.getByText('John Doe')).toBeInTheDocument();
    expect(screen.getByText('Software Engineer')).toBeInTheDocument();
  });

  it('renders QuickAction component', () => {
    renderWithRouter(<UserDashboard />);
    expect(screen.getByText('Quick Actions')).toBeInTheDocument();
  });

  it('renders OverviewSection component', () => {
    renderWithRouter(<UserDashboard />);
    expect(screen.getByText('Available Leave Days')).toBeInTheDocument();
  });
});
