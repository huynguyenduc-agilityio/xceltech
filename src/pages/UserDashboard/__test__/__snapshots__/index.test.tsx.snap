// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDashboard renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <h1
    class="text-2xl text-primary font-bold"
  >
    Dashboard
  </h1>
  <div
    class="relative bg-blue-midnightAzure text-white mt-8 pl-[52px] py-[52px] flex items-center justify-between rounded-regular"
  >
    <div
      class="flex items-center gap-9"
    >
      <div
        class="w-[139px] h-[139px] p-2 rounded-full border-2 border-white"
      >
        <span
          class="relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full text-2xl"
          style="width: 100%; height: 100%;"
        >
          <span
            class="flex h-full w-full items-center justify-center rounded-full text-2xl text-white bg-blue-electric"
          >
            J D
          </span>
        </span>
      </div>
      <div
        class="flex flex-col gap-4"
      >
        <span
          class="text-3xl font-bold"
        >
          John Doe
        </span>
        <span
          class="text-2xl"
        >
          Software Engineer
        </span>
      </div>
    </div>
    <div
      class="flex items-center gap-8"
    >
      <a
        data-discover="true"
        href="/user/dashboard/update-profile"
      >
        <button
          class="font-productsans inline-flex items-center justify-center whitespace-nowrap rounded-[14px] font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-secondary text-black-default hover:bg-secondary/80 py-6 w-[252px] h-[65px] text-xl"
        >
          Edit Profile
        </button>
      </a>
      <img
        alt="profile-icon"
        src="/assets/images/arrow.webp"
      />
    </div>
  </div>
  <div
    class="mt-12 gap-[30px]"
  >
    <h3
      class="text-2xl"
    >
      Quick Actions
    </h3>
    <div
      class="mt-[30px] flex gap-6"
    >
      <a
        class="w-full"
        data-discover="true"
        href="/user/dashboard/leave"
      >
        <button
          class="font-productsans inline-flex items-center justify-center whitespace-nowrap font-bold ring-offset-background duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed h-[70px] w-full py-4 text-xl rounded-full bg-white text-black-soft shadow-lg shadow-black-default/15 hover:bg-secondary transition-all"
        >
          Apply for Leave
        </button>
      </a>
      <button
        class="font-productsans inline-flex items-center justify-center whitespace-nowrap font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:cursor-not-allowed bg-primary text-white h-[70px] w-full py-4 text-xl rounded-full disabled:bg-white disabled:opacity-100 disabled:text-black-soft disabled:hover:bg-white"
        disabled=""
      >
        KPI Goals
      </button>
      <button
        class="font-productsans inline-flex items-center justify-center whitespace-nowrap font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:cursor-not-allowed bg-primary text-white h-[70px] w-full py-4 text-xl rounded-full disabled:bg-white disabled:opacity-100 disabled:text-black-soft disabled:hover:bg-white"
        disabled=""
      >
        Take Appraisal
      </button>
      <button
        class="font-productsans inline-flex items-center justify-center whitespace-nowrap font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:cursor-not-allowed bg-primary text-white h-[70px] w-full py-4 text-xl rounded-full disabled:bg-white disabled:opacity-100 disabled:text-black-soft disabled:hover:bg-white"
        disabled=""
      >
        View Payslip
      </button>
      <button
        class="font-productsans inline-flex items-center justify-center whitespace-nowrap font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:cursor-not-allowed bg-primary text-white h-[70px] w-full py-4 text-xl rounded-full disabled:bg-white disabled:opacity-100 disabled:text-black-soft disabled:hover:bg-white"
        disabled=""
      >
        Update Profile
      </button>
      <button
        class="font-productsans inline-flex items-center justify-center whitespace-nowrap font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:cursor-not-allowed bg-primary text-white h-[70px] w-full py-4 text-xl rounded-full disabled:bg-white disabled:opacity-100 disabled:text-black-soft disabled:hover:bg-white"
        disabled=""
      >
        Events
      </button>
    </div>
  </div>
  <div
    class="grid grid-cols-1 md:grid-cols-2 gap-10 mt-[64px]"
  >
    <div
      class="h-full p-10 rounded-regular bg-white"
    >
      <div
        class="flex items-center justify-between mb-6"
      >
        <span
          class="text-2xl font-bold"
        >
          Available Leave Days
        </span>
        <button
          aria-label="More options"
        >
          <svg
            fill="none"
            height="27"
            width="31"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.5 9.703c2.41 0 4.36 1.698 4.36 3.797s-1.95 3.797-4.36 3.797-4.36-1.698-4.36-3.797 1.95-3.797 4.36-3.797ZM11.14 4.22c0 2.099 1.95 3.797 4.36 3.797s4.36-1.698 4.36-3.797c0-2.1-1.95-3.797-4.36-3.797s-4.36 1.698-4.36 3.797Zm0 18.562c0 2.1 1.95 3.797 4.36 3.797s4.36-1.698 4.36-3.797-1.95-3.797-4.36-3.797-4.36 1.698-4.36 3.797Z"
              fill="#000"
            />
          </svg>
        </button>
      </div>
      <div
        class="flex flex-col gap-7"
      >
        <div
          class="flex flex-col flex-1 w-full"
        >
          <div
            class="flex items-center justify-between mb-3"
          >
            <span
              class="text-lg text-gray-steel"
            >
              Annual Leave
            </span>
            <span
              class="text-lg text-gray-steel"
            >
              10 of 60 day(s)
            </span>
          </div>
          <div
            aria-label="Progress 16"
            aria-valuemax="100"
            aria-valuemin="0"
            class="relative h-7 w-full overflow-hidden rounded-[3px] bg-backgroundGray"
            data-max="100"
            data-state="indeterminate"
            role="progressbar"
          >
            <div
              class="h-full flex-1 transition-all rounded-[3px] bg-primary"
              data-max="100"
              data-state="indeterminate"
              style="transform: translateX(-84%);"
            />
          </div>
        </div>
        <div
          class="flex flex-col flex-1 w-full"
        >
          <div
            class="flex items-center justify-between mb-3"
          >
            <span
              class="text-lg text-gray-steel"
            >
              Sick Leave
            </span>
            <span
              class="text-lg text-gray-steel"
            >
              0 of 10 day(s)
            </span>
          </div>
          <div
            aria-label="Progress 0"
            aria-valuemax="100"
            aria-valuemin="0"
            class="relative h-7 w-full overflow-hidden rounded-[3px] bg-backgroundGray"
            data-max="100"
            data-state="indeterminate"
            role="progressbar"
          >
            <div
              class="h-full flex-1 transition-all rounded-[3px] bg-primary"
              data-max="100"
              data-state="indeterminate"
              style="transform: translateX(-100%);"
            />
          </div>
        </div>
        <div
          class="flex flex-col flex-1 w-full"
        >
          <div
            class="flex items-center justify-between mb-3"
          >
            <span
              class="text-lg text-gray-steel"
            >
              Compassionate Leave
            </span>
            <span
              class="text-lg text-gray-steel"
            >
              8 of 15 day(s)
            </span>
          </div>
          <div
            aria-label="Progress 60"
            aria-valuemax="100"
            aria-valuemin="0"
            class="relative h-7 w-full overflow-hidden rounded-[3px] bg-backgroundGray"
            data-max="100"
            data-state="indeterminate"
            role="progressbar"
          >
            <div
              class="h-full flex-1 transition-all rounded-[3px] bg-primary"
              data-max="100"
              data-state="indeterminate"
              style="transform: translateX(-40%);"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="h-full p-10 rounded-regular bg-white"
    >
      <div
        class="flex items-center justify-between mb-6"
      >
        <span
          class="text-2xl font-bold"
        >
          Birthdays
        </span>
        <button
          aria-label="More options"
        >
          <svg
            fill="none"
            height="27"
            width="31"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.5 9.703c2.41 0 4.36 1.698 4.36 3.797s-1.95 3.797-4.36 3.797-4.36-1.698-4.36-3.797 1.95-3.797 4.36-3.797ZM11.14 4.22c0 2.099 1.95 3.797 4.36 3.797s4.36-1.698 4.36-3.797c0-2.1-1.95-3.797-4.36-3.797s-4.36 1.698-4.36 3.797Zm0 18.562c0 2.1 1.95 3.797 4.36 3.797s4.36-1.698 4.36-3.797-1.95-3.797-4.36-3.797-4.36 1.698-4.36 3.797Z"
              fill="#000"
            />
          </svg>
        </button>
      </div>
      <div
        class="flex flex-col gap-[14px]"
      >
        <div
          class="flex items-center justify-between px-[30px] py-3 bg-blue-light rounded-regular"
        >
          <div
            class="flex items-center gap-6"
          >
            <svg
              class="text-black-default"
              fill="none"
              height="25"
              style="fill: black;"
              viewBox="0 0 23 27"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 24.469C0 25.866 1.104 27 2.464 27h18.072C21.896 27 23 25.866 23 24.469V10.125H0zm16.429-10.336c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm-6.572-6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm-6.571-6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm17.25-17.508H18.07V.844A.835.835 0 0 0 17.25 0h-1.643a.835.835 0 0 0-.821.844v2.531H8.214V.844A.835.835 0 0 0 7.393 0H5.75a.835.835 0 0 0-.821.844v2.531H2.464C1.104 3.375 0 4.509 0 5.906v2.532h23V5.906c0-1.397-1.104-2.531-2.464-2.531"
                fill="currentColor"
              />
            </svg>
            <span
              class="text-lg text-gray-steel"
            >
              Biruk Kidan's Day - April 25th
            </span>
          </div>
          <button
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-secondary text-black-default hover:bg-secondary/80 h-10 rounded-[23px] w-[182px] rounded-regular text-base"
          >
            Send Wishes
          </button>
        </div>
        <div
          class="flex items-center justify-between px-[30px] py-3 bg-blue-light rounded-regular"
        >
          <div
            class="flex items-center gap-6"
          >
            <svg
              class="text-black-default"
              fill="none"
              height="25"
              style="fill: black;"
              viewBox="0 0 23 27"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 24.469C0 25.866 1.104 27 2.464 27h18.072C21.896 27 23 25.866 23 24.469V10.125H0zm16.429-10.336c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm-6.572-6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm-6.571-6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm17.25-17.508H18.07V.844A.835.835 0 0 0 17.25 0h-1.643a.835.835 0 0 0-.821.844v2.531H8.214V.844A.835.835 0 0 0 7.393 0H5.75a.835.835 0 0 0-.821.844v2.531H2.464C1.104 3.375 0 4.509 0 5.906v2.532h23V5.906c0-1.397-1.104-2.531-2.464-2.531"
                fill="currentColor"
              />
            </svg>
            <span
              class="text-lg text-gray-steel"
            >
              Biruk Kidan's Day - April 25th
            </span>
          </div>
          <button
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-secondary text-black-default hover:bg-secondary/80 h-10 rounded-[23px] w-[182px] rounded-regular text-base"
          >
            Send Wishes
          </button>
        </div>
        <div
          class="flex items-center justify-between px-[30px] py-3 bg-blue-light rounded-regular"
        >
          <div
            class="flex items-center gap-6"
          >
            <svg
              class="text-black-default"
              fill="none"
              height="25"
              style="fill: black;"
              viewBox="0 0 23 27"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 24.469C0 25.866 1.104 27 2.464 27h18.072C21.896 27 23 25.866 23 24.469V10.125H0zm16.429-10.336c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm-6.572-6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm-6.571-6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm17.25-17.508H18.07V.844A.835.835 0 0 0 17.25 0h-1.643a.835.835 0 0 0-.821.844v2.531H8.214V.844A.835.835 0 0 0 7.393 0H5.75a.835.835 0 0 0-.821.844v2.531H2.464C1.104 3.375 0 4.509 0 5.906v2.532h23V5.906c0-1.397-1.104-2.531-2.464-2.531"
                fill="currentColor"
              />
            </svg>
            <span
              class="text-lg text-gray-steel"
            >
              Biruk Kidan's Day - April 25th
            </span>
          </div>
          <button
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-secondary text-black-default hover:bg-secondary/80 h-10 rounded-[23px] w-[182px] rounded-regular text-base"
          >
            Send Wishes
          </button>
        </div>
        <div
          class="flex items-center justify-between px-[30px] py-3 bg-blue-light rounded-regular"
        >
          <div
            class="flex items-center gap-6"
          >
            <svg
              class="text-black-default"
              fill="none"
              height="25"
              style="fill: black;"
              viewBox="0 0 23 27"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 24.469C0 25.866 1.104 27 2.464 27h18.072C21.896 27 23 25.866 23 24.469V10.125H0zm16.429-10.336c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm-6.572-6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm-6.571-6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm17.25-17.508H18.07V.844A.835.835 0 0 0 17.25 0h-1.643a.835.835 0 0 0-.821.844v2.531H8.214V.844A.835.835 0 0 0 7.393 0H5.75a.835.835 0 0 0-.821.844v2.531H2.464C1.104 3.375 0 4.509 0 5.906v2.532h23V5.906c0-1.397-1.104-2.531-2.464-2.531"
                fill="currentColor"
              />
            </svg>
            <span
              class="text-lg text-gray-steel"
            >
              Biruk Kidan's Day - April 25th
            </span>
          </div>
          <button
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-secondary text-black-default hover:bg-secondary/80 h-10 rounded-[23px] w-[182px] rounded-regular text-base"
          >
            Send Wishes
          </button>
        </div>
        <div
          class="flex items-center justify-between px-[30px] py-3 bg-blue-light rounded-regular"
        >
          <div
            class="flex items-center gap-6"
          >
            <svg
              class="text-black-default"
              fill="none"
              height="25"
              style="fill: black;"
              viewBox="0 0 23 27"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 24.469C0 25.866 1.104 27 2.464 27h18.072C21.896 27 23 25.866 23 24.469V10.125H0zm16.429-10.336c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632h-2.053a.626.626 0 0 1-.616-.633zm-6.572-6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.054c.339 0 .616.285.616.633v2.11a.627.627 0 0 1-.616.632h-2.054a.626.626 0 0 1-.616-.633zm-6.571-6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm0 6.75c0-.348.277-.633.616-.633h2.053c.34 0 .616.285.616.633v2.11a.626.626 0 0 1-.616.632H3.902a.626.626 0 0 1-.616-.633zm17.25-17.508H18.07V.844A.835.835 0 0 0 17.25 0h-1.643a.835.835 0 0 0-.821.844v2.531H8.214V.844A.835.835 0 0 0 7.393 0H5.75a.835.835 0 0 0-.821.844v2.531H2.464C1.104 3.375 0 4.509 0 5.906v2.532h23V5.906c0-1.397-1.104-2.531-2.464-2.531"
                fill="currentColor"
              />
            </svg>
            <span
              class="text-lg text-gray-steel"
            >
              Biruk Kidan's Day - April 25th
            </span>
          </div>
          <button
            class="font-productsans inline-flex items-center justify-center whitespace-nowrap font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed bg-secondary text-black-default hover:bg-secondary/80 h-10 rounded-[23px] w-[182px] rounded-regular text-base"
          >
            Send Wishes
          </button>
        </div>
      </div>
    </div>
    <div
      class="h-full p-10 rounded-regular bg-white"
    >
      <div
        class="flex items-center justify-between mb-6"
      >
        <span
          class="text-2xl font-bold"
        >
          April Pay Slip Breakdown
        </span>
        <button
          aria-label="More options"
        >
          <svg
            fill="none"
            height="27"
            width="31"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.5 9.703c2.41 0 4.36 1.698 4.36 3.797s-1.95 3.797-4.36 3.797-4.36-1.698-4.36-3.797 1.95-3.797 4.36-3.797ZM11.14 4.22c0 2.099 1.95 3.797 4.36 3.797s4.36-1.698 4.36-3.797c0-2.1-1.95-3.797-4.36-3.797s-4.36 1.698-4.36 3.797Zm0 18.562c0 2.1 1.95 3.797 4.36 3.797s4.36-1.698 4.36-3.797-1.95-3.797-4.36-3.797-4.36 1.698-4.36 3.797Z"
              fill="#000"
            />
          </svg>
        </button>
      </div>
      <div
        class="grid grid-cols-4 text-lg bg-blue-light text-gray-steel px-7 py-4 rounded-regular"
      >
        <div>
          Earnings
        </div>
        <div>
          Amount
        </div>
        <div>
          Deductions
        </div>
        <div>
          Total
        </div>
      </div>
      <div
        class="grid grid-cols-4 text-lg bg-blue-light text-gray-steel px-7 py-4 rounded-regular mt-3"
      >
        <div>
          Basic Wage
        </div>
        <div>
          150,000
        </div>
        <div>
          -30,000
        </div>
        <div>
          120,000
        </div>
      </div>
      <div
        class="grid grid-cols-4 text-lg bg-blue-light text-gray-steel px-7 py-4 rounded-regular mt-3"
      >
        <div>
          Tax
        </div>
        <div>
          15,000
        </div>
        <div>
          -3,000
        </div>
        <div>
          12,000
        </div>
      </div>
      <div
        class="grid grid-cols-4 text-lg bg-blue-light text-gray-steel px-7 py-4 rounded-regular mt-3"
      >
        <div>
          Pension
        </div>
        <div>
          15,000
        </div>
        <div>
          -3,000
        </div>
        <div>
          12,000
        </div>
      </div>
      <div
        class="grid grid-cols-4 text-lg bg-blue-light text-gray-steel px-7 py-4 rounded-regular mt-3"
      >
        <div>
          Total Earnings
        </div>
        <div>
          150,000
        </div>
        <div>
          -36,000
        </div>
        <div>
          114,000
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
